### PR TITLE
Moved some content and added links to Kubeflow docs

### PR DESCRIPTION
**Pull request by sarahmaddox**
**What this PR does / why we need it**: This PR moves some content from the Katib repo to the Kubeflow docs and adds links to the Kubeflow docs, which will soon
contain detailed information about Katib. See companion PR https://github.com/kubeflow/website/pull/1480.

My goal is to centralize the Katib user docs, so that people can find them more easily, and to make Katib easier to use by showing people how it works within Kubeflow.

**Which issue(s) this PR fixes** :
Related to issue https://github.com/kubeflow/website/issues/1425

**Special notes for your reviewer**:

1. *Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.*
No change to image versions.

1. Hold this PR until PR https://github.com/kubeflow/website/pull/1480 is merged.

1. Companion PR https://github.com/kubeflow/website/pull/1480 adds detailed information about Katib to the Kubeflow docs. You can see the preview of the new docs here:

    * https://deploy-preview-1480--competent-brattain-de2d6d.netlify.com/docs/components/hyperparameter-tuning/
    * https://deploy-preview-1480--competent-brattain-de2d6d.netlify.com/docs/components/hyperparameter-tuning/overview/
    * https://deploy-preview-1480--competent-brattain-de2d6d.netlify.com/docs/components/hyperparameter-tuning/hyperparameter/
    * https://deploy-preview-1480--competent-brattain-de2d6d.netlify.com/docs/components/hyperparameter-tuning/experiment/

**Release note**:

```release-note
NONE
```

This PR is now ready for review. Applying a hold to allow for multiple reviewers.
/hold

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/990)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
/assign @gaocegege
/assign @hougangliu
/assign @richardsliu
/cc @abhi-g 

Please would you review this PR and its companion PR https://github.com/kubeflow/website/pull/1480. My goal is to centralize the Katib user docs, so that people can find them more easily, and to make Katib easier to use by showing people how it works within Kubeflow.

File: README.md [hunk 1]
```diff
@@ -34,12 +34,22 @@ The system is inspired by [Google vizier][3] and supports multiple ML/DL framewo
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+## Getting Started
```
Comment (thedriftofwords) [inline README.md]: Nit: Update titles to sentence case

File: README.md [hunk 2]
```diff
@@ -34,12 +34,22 @@ The system is inspired by [Google vizier][3] and supports multiple ML/DL framewo
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
```
Comment (thedriftofwords) [inline README.md]: Not sure where the source of the TOC is, but on line 29: Pytorch --> PyTorch

File: docs/workflow-design.md [hunk 3]
```diff
@@ -66,17 +74,12 @@ status:
...
-- **maxTrialCount**: It specifies how many sets of hyperparameter can be generated to test the model at most.
-- **parallelTrialCount**: This fields specifies how many sets of hyperparameter to be tested in parallel at most.
-- **maxFailedTrialCount**: Some sets of hyperparameter corresponding jobs maybe fail somehow. If the failed count of hyperparameter set exceeds `maxFailedTrialCount`, the hyperparameter tuning for the model will be stopped with `Failed` status.
+When you want to tune hyperparameter for your machine learning model before 
+training it further, you just need create an `Experiment` CR like above. To
```
Comment (thedriftofwords) [inline docs/workflow-design.md]: ...you just need *to* create...

File: docs/workflow-design.md [hunk 4]
```diff
@@ -66,17 +74,12 @@ status:
...
+lear what fields are included in the `Experiment.spec`, see
```
Comment (thedriftofwords) [inline docs/workflow-design.md]: learn

File: docs/workflow-design.md [hunk 5]
```diff
@@ -66,17 +74,12 @@ status:
...
-- **maxTrialCount**: It specifies how many sets of hyperparameter can be generated to test the model at most.
-- **parallelTrialCount**: This fields specifies how many sets of hyperparameter to be tested in parallel at most.
-- **maxFailedTrialCount**: Some sets of hyperparameter corresponding jobs maybe fail somehow. If the failed count of hyperparameter set exceeds `maxFailedTrialCount`, the hyperparameter tuning for the model will be stopped with `Failed` status.
+When you want to tune hyperparameter for your machine learning model before
```
Comment (thedriftofwords) [inline docs/workflow-design.md]: hyperparameters

**Comment by sarahmaddox** [comment]
Thanks @thedriftofwords for the tech writer review! Since the comments are about typos, and since all except one are pre-existing typos, I'll let this PR go thru as is and I'll submit a followup PR to fix the typos.

Just waiting on a final LGTM on the related https://github.com/kubeflow/website/pull/1480, then this one can go through too.

**Pull request merged**